consensus/src/dag/dag_driver.rs

```diff
@@ -139,14 +140,31 @@ impl DagDriver {
 
     pub async fn enter_new_round(&mut self, new_round: Round, strong_links: Vec<NodeCertificate>) {
         debug!("entering new round {}", new_round);
-        // TODO: support pulling payload
+        let payload_filter = {
+            let dag_reader = self.dag.read();
+            let highest_commit_round = dag_reader.highest_committed_anchor_round();
+            if strong_links.is_empty() {
+                PayloadFilter::Empty
+            } else {
+                PayloadFilter::from(
+                    &dag_reader
+                        .reachable(
+                            strong_links.iter().map(|node| node.metadata()),
+                            Some(highest_commit_round.saturating_sub(DAG_WINDOW as u64)),
```

Comment:
What if proofs in nodes outside of the window are not committed yet?

Reply:
they're not going to be committed ever by the rule right? (if you mean the nodes with round < highest_committed_round - WINDOW)

Reply:
Yes, what I mean is that QS can give it the DAG because they were not committed yet (decoupled execution).

Reply:
that's not possible right? the highest_commit round is what already goes to storage, not the highest ordered round

Reply:
I am confused. I think the window has to be according to the highest-ordered round. Otherwise, it is not deterministic.

Reply:
Sorry, I think you are good. I was confused.

Reply:
ah yes, we haven't done that for order rule, but here we're just being more relaxed? since commit round <= ordered round

Reply:
It is tricky. 
With the current rule, we might filter out proof that we need to resubmit (because their nodes are below the GC line).
But if we do "ordered round - window", we might include stuff that we are already excited but not yet committed.

Reply:
it's fine to filter out proof that need to resubmit since the window is moving, it's just delaying those proof for a little bit?

Reply:
What about recording?
Also, they can expire.